sdk/pinocchio/src/account_info.rs

```diff
@@ -376,6 +376,16 @@ impl AccountInfo {
             }
         }
 
+        #[cfg(not(target_os = "solana"))]
+        if zero_init {
+            let len_increase = new_len.saturating_sub(current_len);
+            if len_increase > 0 {
+                unsafe {
+                    core::ptr::write_bytes(data.as_mut_ptr().add(current_len), 0, len_increase);
```

Comment:
You can add this under the `sol_memset_` call:
```rust
#[cfg(not(target_os = "solana"))]
 core::ptr::write_bytes(data.as_mut_ptr().add(current_len), 0, len_increase);
 ```
The other bits of duplicated code can be removed.